### PR TITLE
Shipment status

### PR DESCRIPTION
**Pull request by jakrzus**
### As an admin, I want to be able to update the order's shipment status so it can be completed.

- [x] New orders have a "pending" shipment status
- [x] Admin can change an order's shipment status using a button on the order's page
- [x] Shipment status can be changed from "pending" to "ready" or "canceled"
- [x] Shipment status can be changed from "ready" to "shipped" only if its payment status is "completed"
- [x] Shipment status can be changed from "ready" to "failed"
- [x] There is a system & request spec

##

Added JSON response to the shipment controller

### CODE REVIEW

File: app/controllers/admin/shipments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    end
+
+    private
+
+    def event
+      params[:event]
+    end
+
+    def shipment
+      Shipment.find params[:id]
```
Comment (kamilpuchala) [inline app/controllers/admin/shipments_controller.rb:25]: I think you should use here memoization

File: db/migrate/20210731154953_create_shimpent_objects.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class CreateShimpentObjects < ActiveRecord::Migration[6.1]
```
Comment (kamilpuchala) [inline db/migrate/20210731154953_create_shimpent_objects.rb:1]: I'll suggest to do it in a rake task, which will be called only 1 time

File: spec/requests/admin_shipments_spec.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require './spec/support/helpers'
+
+RSpec.describe 'AdminShipments', type: :request do
+  let!(:admin) { create :admin_user }
```
Comment (kamilpuchala) [inline spec/requests/admin_shipments_spec.rb:7]: Can you add minimum one spec for a user which isn't an admin?

File: spec/system/admin_orders_spec.rb [hunk 4]
```diff
@@ -47,4 +47,23 @@
 
     expect(page).to have_content 'Payment status: completed'
   end
+
+  it 'allows admin to change shipment status to shipped' do
```
Comment (kamilpuchala) [inline spec/system/admin_orders_spec.rb:51]: Here the same suggestion as in previous comment

Comment (kamilpuchala) [inline spec/system/admin_orders_spec.rb:51]: Same here as in previous comment

**Review by kamilpuchala — APPROVED** [review]
It looks nice and clean. GJ, please check my comments, but all of them are mostly small improvements

**Review by kamilpuchala — APPROVED** [review]
I think it’s  good enough to give you an approve so just consider my comments :)

File: spec/system/admin_orders_spec.rb (hunk 4 above)
Comment (jakrzus) [reply spec/system/admin_orders_spec.rb:51]: I'm not really sure how to set such spec up. If the user is not an admin, then visiting the admin root path there will be a redirection to the login page, and there is already an existing spec for this.
```
it 'does not allow non admin user to login to admin dashboard' do
    click_on 'Sign In'
    fill_in_sign_in_form_as user
    click_on 'Log in'
    visit admin_root_path

    expect(page).to have_content('You are not authorized')
  end
```
  Or should I do something like `expect(page).not_to have_css '#orders-table'`?  But this does not look like it belongs to a system spec.

Comment (jakrzus) [reply spec/system/admin_orders_spec.rb:51]: Maybe i should test `expect(page).not_to have_css '#orders-table'` in the request spec?

Comment (kamilpuchala) [reply spec/system/admin_orders_spec.rb:51]: We discussed this on slack :)

**Review by kamilpuchala — APPROVED** [review]

**Pull request merged**